components/fxa-client/src/oauth.rs

```diff
@@ -163,20 +163,67 @@ impl FirefoxAccount {
     /// * `state` - OAuth state.
     /// * `access_type` - Type of OAuth access, can be "offline" and "online.
     pub fn authorize_code_using_session_token(
-        &self,
+        // Is mut purely for the sake of this draft
+        // Will be removed
+        &mut self,
         client_id: &str,
         scope: &str,
         state: &str,
         access_type: &str,
     ) -> Result<String> {
         let session_token = self.get_session_token()?;
+        let mut code_challenge = None;
+        let mut keys_jwe = None;
+        let mut code_challenge_method = None;
+        // Is this check even neccesary? Can we just compute the keys_jwe and
+        // Send it anyway, filtering the self.state.scoped_keys on the reqeusted scopes
```

Comment:
Do you mean that we would always create a `keys_jwe`, but the items that it contains would be filetered based on `scope`? (And so if the `OLD_SYNC` scope was not requested, it would be a JWE containing the string `"{}"`?). That...sounds OK to me, if it works out to be simpler to implement.